### PR TITLE
Fix netcdf datetime handling

### LINKED ISSUE #115: Incorrect times after `
```python

### PR DESCRIPTION
**Pull request by gadomski**
Fixes netcdf datetime handling by removing the fill value if the dataset is a netcdf datetime. NetCDF datetimes are detected using the same heuristic that xarray uses: https://github.com/pydata/xarray/blob/f8bae5974ee2c3f67346298da12621af4cae8cf8/xarray/coding/times.py#L670.

## Related issues
- Closes #115 

## Additional changes
- Adds more required dependencies to `requirements.txt`
- Adds a `requirements-dev.txt` with development dependencies, for developers that aren't using conda

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,9 @@
+cftime
```
Comment (martindurant) [inline requirements.txt:1]: Please do not add all thee requirements. If we wanted to kerchunk FITS files, for example, we wouldn't need all these. You could add them to optional install selectors, to install like `pip install kerchunk[cdf,grib]`...

File: kerchunk/hdf.py [hunk 2]
```diff
@@ -302,3 +304,10 @@ def _storage_info(self, dset: h5py.Dataset) -> dict:
                     [a // b for a, b in zip(blob.chunk_offset, chunk_size)])
                 stinfo[key] = {'offset': blob.byte_offset, 'size': blob.size}
             return stinfo
+
+
+def _is_netcdf_datetime(dataset: h5py.Dataset):
+    attrs = dataset.attrs
+    # This is the same heuristic used by xarray
+    # https://github.com/pydata/xarray/blob/f8bae5974ee2c3f67346298da12621af4cae8cf8/xarray/coding/times.py#L670
+    return "units" in attrs and "since" in attrs["units"]
```
Comment (martindurant) [inline kerchunk/hdf.py:313]: Should be `attrs.get("units", "")`, since the attribute might not exist.

**Review by martindurant — COMMENTED** [review]
This seems like a reasonable thing to do for this very specific case. It is interesting that xarray uses this logic without invoking cftime directly.

Note that this case is already dealt with at combine time, by specifying that the given input data types are to be interpreted using cftime. This is done by adding, for example `{"time": "cf:time"}` to the coordinate mapper, which says "calculate the coordinate values of 'time' by applying cftime to the data variable called 'time'"

File: kerchunk/hdf.py (hunk 2 above)
Comment (gadomski) [reply kerchunk/hdf.py:313]: Shouldn't the initial `"units" in attrs` handle that case?

Comment (martindurant) [reply kerchunk/hdf.py:313]: Sorry, I'm talking rubbish, ignore me :)

Comment (gadomski) [reply kerchunk/hdf.py:313]: No worries :-)

File: requirements.txt [hunk 3]
```diff
@@ -1,2 +1,6 @@
+cfgrib
```
Comment (martindurant) [inline requirements.txt:1]: cfgrib is also strictly optional

Comment (gadomski) [reply requirements.txt:1]: Fixed.

**Comment by martindurant** [comment]
👍

**Pull request merged**